### PR TITLE
(feat) O3-2485: Allow view only field for visit location in the start visit form

### PR DESCRIPTION
**Pull request by vasharma05**
## Requirements

- [x] This PR has a title that briefly describes the work done including the ticket number. If there is a ticket, make sure your PR title includes a [conventional commit](https://o3-dev.docs.openmrs.org/#/getting_started/contributing?id=your-pr-title-should-indicate-the-type-of-change-it-is) label. See existing PR titles for inspiration.
- [x] My work conforms to the [OpenMRS 3.0 Styleguide](https://om.rs/styleguide) and [design documentation](https://zeroheight.com/23a080e38/p/880723-introduction).
- [x] My work includes tests or is validated by existing tests.

## Summary

This PR adds a new configuration point to the Patient Chart's configuration schema. This property determines whether the `Visit location` field in the Start Visit form is editable or read-only. By default, the value of the field gets resolved to the value of the logged-in user's `sessionLocation` property.

Sample configuration (set to `false` so the input is editable by default):

```
viewOnlyVisitLocationField: {
  _type: Type.Boolean,
  _description: 'Whether the visit location field in the start visit form should be view-only',
  _default: false,
}
```

## Screenshots

image

## Related Issue

https://issues.openmrs.org/projects/O3/issues/O3-2485

## Other

*None*

### CODE REVIEW

File: packages/esm-patient-chart-app/src/config-schema.ts [hunk 1]
```diff
@@ -112,6 +112,11 @@ export const esmPatientChartSchema = {
       'An array of concept UUIDs. If an observation has a concept UUID that matches any of the ones in this array, it will be hidden from the observations list in the Encounters summary table.',
     _default: [],
   },
+  viewOnlyVisitLocationField: {
```
Comment (denniskigen) [inline packages/esm-patient-chart-app/src/config-schema.ts:115]: Pedantic, but perhaps `readOnlyVisitLocationField` might be a better fit here. Additionally, the `ComboBox` component provides a [readOnly](https://react.carbondesignsystem.com/?path=/story/components-combobox--playground&args=disabled:!false;readOnly:!true) boolean prop. Might that be preferable to using `disabled`?

File: packages/esm-patient-flags-app/translations/es.json [hunk 2]
```diff
@@ -10,6 +10,7 @@
   "editFlags": "Editar banderas",
   "enabledFlag": "Bandera activada",
   "enablingFlag": "Activando bandera...",
+  "flagCountfalsemany": "{count} risk flag{plural}",
```
Comment (denniskigen) [inline packages/esm-patient-flags-app/translations/es.json:13]: Does #1413 solve the issue of this specific translation randomly showing up in unrelated diffs? Perhaps rebasing against `main` might fix it?

**Review by denniskigen — APPROVED** [review]
LGTM with a few pedantic concerns.

File: packages/esm-patient-flags-app/translations/es.json (hunk 2 above)
Comment (vasharma05) [reply packages/esm-patient-flags-app/translations/es.json:13]: This actually is the wrong key, instead of false, it should be _

Comment (vasharma05) [reply packages/esm-patient-flags-app/translations/es.json:13]: I'LL fix this asap

File: packages/esm-patient-chart-app/src/visit/visit-form/location-selection.component.tsx [hunk 3]
```diff
@@ -34,24 +38,29 @@ const LocationSelector = () => {
...
+                aria-label={t('selectLocation', 'Select a location')}
+                id="location"
+                invalidText="Required"
+                items={locationsToShow}
+                selectedItem={value}
+                onChange={({ selectedItem }) => onChange(selectedItem)}
+                onBlur={onBlur}
+                itemToString={(loc: Location) => loc?.display}
+                onInputChange={(loc) => handleSearch(loc)}
+                disabled={viewOnlyVisitLocationField}
```
Comment (brandones) [inline packages/esm-patient-chart-app/src/visit/visit-form/location-selection.component.tsx:57]: @vasharma05 I agree with @denniskigen that `readOnly` is more suitable here.

```suggestion
                readOnly={viewOnlyVisitLocationField}
```

File: packages/esm-patient-chart-app/src/config-schema.ts (hunk 1 above)
Comment (brandones) [reply packages/esm-patient-chart-app/src/config-schema.ts:115]:
```suggestion
  disableChangingVisitLocation: {
```

File: packages/esm-patient-chart-app/src/config-schema.ts [hunk 4]
```diff
@@ -112,6 +112,11 @@ export const esmPatientChartSchema = {
...
+    _type: Type.Boolean,
+    _description: 'Whether the visit location field in the start visit form should be view only',
```
Comment (brandones) [inline packages/esm-patient-chart-app/src/config-schema.ts:117]:
```suggestion
    _description: 'Whether the visit location field in the Start Visit form should be view-only. If so, the visit location will always be set to the user's login location.',
```

**Review by brandones — CHANGES_REQUESTED** [review]

**Review by brandones — APPROVED** [review]

File: packages/esm-patient-chart-app/src/visit/visit-form/location-selection.component.tsx [hunk 5]
```diff
@@ -34,24 +38,29 @@ const LocationSelector = () => {
...
-              items={locationsToShow}
-              selectedItem={value}
-              onChange={({ selectedItem }) => onChange(selectedItem)}
-              onBlur={onBlur}
-              itemToString={(loc: Location) => loc?.display}
-              onInputChange={(loc) => handleSearch(loc)}
-            />
-          )}
-        />
+        {!viewOnlyVisitLocationField ? (
```
Comment (vasharma05) [inline packages/esm-patient-chart-app/src/visit/visit-form/location-selection.component.tsx:41]: Hi @brandones !
I see that the config prop was renamed but it wasn't updated in here.
I'll create a new PR for this.

Thank you for the changes and review @brandones @denniskigen !

Comment (brandones) [reply packages/esm-patient-chart-app/src/visit/visit-form/location-selection.component.tsx:41]: Oh no, I'm sorry I missed that. My bad. Thank you Vineet.
